### PR TITLE
Wpcs compliance

### PR DESCRIPTION
**Pull request by pinalj**

### CODE REVIEW

File: woocommerce-ac.php [hunk 1]
```diff
@@ -3019,164 +3286,175 @@ function wcal_show_help_tips() {
...
+											print '<input type="checkbox" name="is_wc_template" id="is_wc_template" ' . esc_attr( $is_wc_template ) . '>  </input>';
+											?>
+											<img class="help_tip" width="16" height="16" data-tip='<?php esc_html_e( 'Use WooCommerce default style template for abandoned cart reminder emails.', 'woocommerce' ); ?>' src="<?php echo esc_url( WC()->plugin_url() ); ?>/assets/images/help.png" /><a target = '_blank' href= <?php echo esc_url( wp_nonce_url( admin_url( "?wcal_preview_woocommerce_mail=true&id=$edit_id" ), 'woocommerce-abandoned-cart' ) ); ?> >
+											Click here to preview </a>how the email template will look with WooCommerce Template Style enabled. Alternatively, if this is unchecked, the template will appear as <a target = '_blank' href=<?php echo wp_nonce_url( admin_url( "?wcal_preview_mail=true&id=$edit_id" ), 'woocommerce-abandoned-cart' ); ?>>shown here</a>. <br> <strong>Note: </strong>When this setting is enabled, then "Send From This Name:" & "Send From This Email Address:" will be overwritten with WooCommerce -> Settings -> Email -> Email Sender Options.
```
Comment (dhruvins) [inline woocommerce-ac.php:3308]: esc_url() must be used for URL

File: woocommerce-ac.php [hunk 2]
```diff
@@ -1151,189 +1232,248 @@ function wcal_store_cart_timestamp() {
...
-				if ( $gdpr_consent ) {
-					$query   = "SELECT * FROM `".$wpdb->prefix."ac_abandoned_cart_history_lite` WHERE user_id = %d AND cart_ignored = '0' AND recovered_cart = '0' AND user_id != '0'";
-					$results = $wpdb->get_results( $wpdb->prepare( $query, $user_id ) );
+				if ( $gdpr_consent && $user_id > 0 ) {
+					$results = $wpdb->get_results( //phpcs:ignore
+						$wpdb->prepare(
+							'SELECT * FROM `' . $wpdb->prefix . 'ac_abandoned_cart_history_lite` WHERE user_id = %d AND cart_ignored = %d AND recovered_cart = %d',
+							$user_id,
+							0,
+							0,
```
Comment (dhruvins) [inline woocommerce-ac.php:1354]: extra comma at the end of the line causing syntax error

File: woocommerce-ac.php [hunk 3]
```diff
@@ -1151,189 +1232,248 @@ function wcal_store_cart_timestamp() {
...
+							'SELECT * FROM `' . $wpdb->prefix . 'ac_abandoned_cart_history_lite` WHERE user_id = %d AND cart_ignored = %s AND recovered_cart = %d',
+							$user_id,
+							$cart_ignored,
+							$recovered_cart
+						)
+					);
+					if ( 0 === count( $results ) ) {
 						$cart_info_meta         = array();
 						$cart_info_meta['cart'] = WC()->session->cart;
 						$cart_info_meta         = json_encode( $cart_info_meta );
```
Comment (dhruvins) [inline woocommerce-ac.php:1258]: wp_json_encode must be used

File: woocommerce-ac.php [hunk 4]
```diff
@@ -1151,189 +1232,248 @@ function wcal_store_cart_timestamp() {
...
 							$abandoned_cart_id = $wpdb->insert_id;
 							wcal_common::wcal_set_cart_session( 'abandoned_cart_id_lite', $abandoned_cart_id );
 						}
 					} elseif ( isset( $results[0]->abandoned_cart_time ) && $compare_time > $results[0]->abandoned_cart_time ) {
-						//$wcal_woocommerce_persistent_cart = version_compare( $woocommerce->version, '3.1.0', ">=" ) ? '_woocommerce_persistent_cart_' . get_current_blog_id() : '_woocommerce_persistent_cart' ;
-						//$updated_cart_info                = json_encode( get_user_meta( $user_id, $wcal_woocommerce_persistent_cart, true ) );
-
 						$updated_cart_info         = array();
 						$updated_cart_info['cart'] = WC()->session->cart;
 						$updated_cart_info         = json_encode( $updated_cart_info );
```
Comment (dhruvins) [inline woocommerce-ac.php:1279]: wp_json_encode to be used

File: woocommerce-ac.php [hunk 5]
```diff
@@ -1151,189 +1232,248 @@ function wcal_store_cart_timestamp() {
...
-							update_user_meta ( $user_id, '_woocommerce_ac_modified_cart', md5( "no" ) );
+							update_user_meta( $user_id, '_woocommerce_ac_modified_cart', md5( 'no' ) );
 						}
 					} else {
-						//$wcal_woocommerce_persistent_cart = version_compare( $woocommerce->version, '3.1.0', ">=" ) ? '_woocommerce_persistent_cart_' . get_current_blog_id() : '_woocommerce_persistent_cart' ;
-						//$updated_cart_info                = json_encode( get_user_meta( $user_id, $wcal_woocommerce_persistent_cart, true ) );
-
 						$updated_cart_info         = array();
 						$updated_cart_info['cart'] = WC()->session->cart;
 						$updated_cart_info         = json_encode( $updated_cart_info );
```
Comment (dhruvins) [inline woocommerce-ac.php:1311]: wp_json_encode to be used

**Review by dhruvins — CHANGES_REQUESTED** [review]
wcal_class-guest.php - we must add unslash to other fields as well

File: woocommerce-ac.php (hunk 1 above)
Comment (pinalj) [reply woocommerce-ac.php]: Fixed.

File: woocommerce-ac.php (hunk 2 above)
Comment (pinalj) [reply woocommerce-ac.php]: Fixed.

File: woocommerce-ac.php (hunk 3 above)
Comment (pinalj) [reply woocommerce-ac.php]: Fixed

File: woocommerce-ac.php (hunk 4 above)
Comment (pinalj) [reply woocommerce-ac.php]: Fixed

File: woocommerce-ac.php (hunk 5 above)
Comment (pinalj) [reply woocommerce-ac.php]: Fixed

**Pull request merged**